backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -289,7 +291,42 @@ object Compiler {
       // Sources are all files
       val sources = inputs.sources.map(path => converter.toVirtualFile(path.underlying))
       val classpath = inputs.classpath.map(path => converter.toVirtualFile(path.underlying))
-      val optionsWithoutFatalWarnings = inputs.scalacOptions.flatMap { option =>
+      def existsReleaseSetting = inputs.scalacOptions.exists(opt =>
+        opt.startsWith("-release") ||
+          opt.startsWith("--release") ||
+          opt.startsWith("-java-output-version")
+      )
+      def sameHome = inputs.javacBin match {
+        case Some(bin) => bin.getParent.getParent == JavaRuntime.home
+        case None => false
+      }
+
+      val scalacOptions = inputs.javacBin.flatMap(binary =>
+        // <JAVA_HOME>/bin/java
+        JavaRuntime.getJavaVersionFromJavaHome(binary.getParent.getParent)
```

Comment:
why is checking java home relevant? I thought that what's important is checking current java version of the bloop process?

Reply:
This is the java version of Java we are compiling with, not the Java version Bloop is using. The latter is below in line 313  `  val bloopNumVer = JavaRuntime.version.takeWhile(_.isDigit).toInt`